### PR TITLE
#153740731 Paystack payment integration

### PR DESCRIPTION
**Pull request by HamdalahA**
#### What does this PR do?
- Integrate Paystack as a payment method

#### Description of Task to be completed?
- Integrate paystack API
- Integrate paystack payment for client users

#### How should this be manually tested?
- Add product to cart and on checkout, the paystack option will be available.

Any background context you want to provide?
- None
What are the relevant pivotal tracker stories
- https://www.pivotaltracker.com/story/show/153740731

#### Screenshots (if appropriate)
screen shot 2018-01-14 at 21 16 38

screen shot 2018-01-14 at 18 53 14
screen shot 2018-01-14 at 21 39 25

### CODE REVIEW

**Review by larrystone — APPROVED** [review]
All good, please take out time to fix the eof newline warnings

**Review by proxiex — APPROVED** [review]

File: .eslintignore [hunk 1]
```diff
@@ -5,3 +5,4 @@ server/plugins.js
 client/plugins.js
 packages/*
 .reaction/jsdoc/templates/*
+imports/plugins/included/payments-paystack/lib/api/paystackLib.js
```
Comment (monsieurkayode) [inline .eslintignore]: Fix this eol error

File: imports/plugins/included/default-theme/client/styles/payment/payment-paystack-checkout.less [hunk 2]
```diff
@@ -0,0 +1,6 @@
+.button-wrapper{
+    margin-bottom:14px;
+    }
+#paystackCheckoutForm{
+    margin-top: 14px;
+    }
```
Comment (monsieurkayode) [inline imports/plugins/included/default-theme/client/styles/payment/payment-paystack-checkout.less]: Fix eol here

**Review by chibuezeayogu — APPROVED** [review]
Good job.

File: imports/plugins/included/payments-paystack/client/checkout/paystack.html [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<template name="paystackPaymentForm">
```
Comment (monsieurkayode) [inline imports/plugins/included/payments-paystack/client/checkout/paystack.html:1]: There are inconsistencies with quote style used in this file

File: imports/plugins/included/payments-paystack/client/checkout/paystack.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        const amount = Math.round(cart.cartTotal()) * 100;
+        const template = this.template;
+        const details = {
+          key: publicKey,
+          name: doc.payerName,
+          email: doc.payerEmail,
+          reference: Random.id(),
+          amount,
+          callback(response) {
+          // const secretKey = secretKey;
```
Comment (monsieurkayode) [inline imports/plugins/included/payments-paystack/client/checkout/paystack.js]: Remove all console log

File: imports/plugins/included/payments-paystack/client/settings/components/paystackSettingsForm.js [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+          <div>
+            <Checkbox
+              label="Authorize"
+              onChange={this.handleCheckBox}
+              name="Authorize"
+              checked={this.state.checkbox.Authorize}
+            />
+          </div>
+
+          {/* <div>
```
Comment (monsieurkayode) [inline imports/plugins/included/payments-paystack/client/settings/components/paystackSettingsForm.js]: Remove commented code

File: imports/plugins/included/payments-paystack/client/settings/containers/paystackSettingsFormContainer.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+  }
+
+  handleChange(e) {
+    e.preventDefault();
+    this.setState({ secretKey: e.target.value });
+    this.setState({ privateKey: e.target.value });
+  }
+
+  handleSubmit(settings) {
+    // e.preventDefault();
```
Comment (monsieurkayode) [inline imports/plugins/included/payments-paystack/client/settings/containers/paystackSettingsFormContainer.js:31]: Remove commented code

File: imports/plugins/included/payments-paystack/lib/collections/schemas/paystack.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
+//
```
Comment (monsieurkayode) [inline imports/plugins/included/payments-paystack/lib/collections/schemas/paystack.js]: Fix this

File: imports/plugins/included/payments-paystack/register.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+        "Authorize",
+        "Capture",
+        "Refund"
+      ]
+    }
+  },
+  registry: [
+    // Settings panel
+    {
+      label: "Paystack Payment", // this key (minus spaces) is used for translations
```
Comment (monsieurkayode) [inline imports/plugins/included/payments-paystack/register.js]: Fix inline comment here

File: .eslintignore (hunk 1 above)
Comment (HamdalahA) [reply .eslintignore]: All fixed! Thanks

**Comment by andela-moe** [comment]
Please review this. You have failing test on circle CI

**Pull request closed without merging**